src/api.py

```diff
@@ -0,0 +1,28 @@
+import requests 
+import hashlib
+
+
+class response:
+    def __init__(self, filename, type_of_data, size, url, checksum):
+        self.filename = filename
+        self.type_of_data = type_of_data
+        self.size=size
+        self.url=url
+        self.checksum=checksum
+
+
+def fetch(url):
+    try:
+        con = requests.head(url)
+        r = requests.get(url)
+        for i in range(len(url.split('/'))-1):
+            name = url.split('/')[-(i+1)]
+            if not(name =='' ):
+                break
+        filename = '../download/'+name
```

Comment:
you need to use a `tempfile` OR alternatively you accept an argument with the path of where to save things but never something with a hardcoded path.